pkg/hintrunner/hinter/scope.go

```diff
@@ -65,14 +86,14 @@ func (sm *ScopeManager) AssignVariables(values map[string]any) error {
 	return nil
 }
 
-func (sm *ScopeManager) GetVariableValue(name string) (any, error) {
+func (sm *ScopeManager) GetVariableValue(name string) (*ScopeValue, error) {
```

Comment:
The struct you proposed looks like this:
```type ScopeMap map[string]ScopeValue```
If you save the pointer to the ScopeValue in a ```value``` variable and return the pointer to the ```value```, that might cause dangling pointer and unneccesary heap allocation. It would be better to just return the value.

Reply:
I already create other function to pass the BigInt and Felt value directly

Reply:
This function is ok, but the case is not to return a pointer created inside this function, but to return ScopeValue as value. This should reduce the heap allocations.
Do you mean that this function is no longer used? If so, then it can be removed. Let me know 😁

Reply:
nvm got it haha